iceoryx_examples/icedelivery/README.md

```diff
@@ -242,25 +227,62 @@ The typed publisher application is an example for a high-level user API and does
 described before. In this summary, just the differences to the prior publisher application are described.
 
 Starting again with the includes, there is now a different one:
+
 ```cpp
 #include "iceoryx_posh/popo/publisher.hpp"
 ```
 
 When it comes to the runtime, things are the same as in the untyped publisher. However, a typed publisher object is
-created
+created and the transmitted data type is provided as template parameter:
+
 ```cpp
 iox::popo::Publisher<RadarObject> publisher({"Radar", "FrontLeft", "Object"});
 ```
 
+The topic type must be default- and copy-constructible when the typed API is used.
```

Comment:
- The topic type does not need to be default constructible. If you have a non default ctor you can provide all arguments to `publisher.load(arg1, arg2 ...)` and they will be forwarded to the constructor (like emplace in the optional).

- The topic type must be trivially copyable: This type traits has to return true: https://en.cppreference.com/w/cpp/types/is_trivially_copyable.
- As an additionaly description you could state that this means that the data is memcpy-able!

Reply:
@elfenpiff I agree with you at point one. Regarding your second point: if we force that we would not be able to send for instance the cxx::vector

Reply:
Removed